src/components/policy/policy_regular/src/sql_pt_representation.cc

```diff
@@ -1177,9 +1177,11 @@ bool SQLPTRepresentation::SaveConsumerFriendlyMessages(
   // the policy table. So it won't be changed/updated
   if (messages.messages.is_initialized()) {
     utils::dbms::SQLQuery query(db());
-    if (!query.Exec(sql_pt::kDeleteMessageString)) {
-      LOG4CXX_WARN(logger_, "Incorrect delete from message.");
-      return false;
+    if (!messages.messages->empty()) {
```

Comment:
@dev-gh I've done some refactoring for this function for both cc-files. See https://github.com/smartdevicelink/sdl_core/pull/1615/commits/045a313b4d99b6141ae99e71ff0e9d1d20520a1c